object-cache.php

```diff
@@ -354,68 +360,103 @@ function get( $id, $group = 'default', $force = false, &$found = null ) {
-			$mc =& $this->get_mc( $group );
-			$keys = array();
-			$this->timer_start();
+		foreach ( $ids as $id ) {
+			$key = $this->key( $id, $group );
 
-			foreach ( $ids as $id ) {
-				$key = $this->key( $id, $group );
-				$keys[] = $key;
-
-				if ( isset( $this->cache[ $key ] ) ) {
-					if ( is_object( $this->cache[ $key ][ 'value'] ) ) {
-						$return[ $key ] = clone $this->cache[ $key ][ 'value'];
-						$return_cache[ $key ] = [
-							'value' => clone $this->cache[ $key ][ 'value'],
-							'found' => $this->cache[ $key ][ 'found'],
-						];
-					} else {
-						$return[ $key ] = $this->cache[ $key ][ 'value'];
-						$return_cache[ $key ] = [
-							'value' => $this->cache[ $key ][ 'value' ],
-							'found' => $this->cache[ $key ][ 'found' ],
-						];
-					}
-
-					continue;
-				} else if ( in_array( $group, $this->no_mc_groups ) ) {
-					$return[ $key ] = false;
+			if ( isset( $this->cache[ $key ] ) && ( ! $force || $no_mc ) ) {
+				if ( is_object( $this->cache[ $key ][ 'value' ] ) ) {
+					$return[ $id ] = clone $this->cache[ $key ][ 'value' ];
 					$return_cache[ $key ] = [
-						'value' => false,
-						'found' => false,
+						'value' => clone $this->cache[ $key ][ 'value' ],
+						'found' => $this->cache[ $key ][ 'found' ],
 					];
-
-					continue;
 				} else {
-					$fresh_get = $mc->get( $key );
-					$return[ $key ] = $fresh_get;
+					$return[ $id ] = $this->cache[ $key ][ 'value' ];
 					$return_cache[ $key ] = [
-						'value' => $fresh_get,
-						'found' => false !== $fresh_get,
+						'value' => $this->cache[ $key ][ 'value' ],
+						'found' => $this->cache[ $key ][ 'found' ],
 					];
 				}
+
+				continue;
+			} else if ( $no_mc ) {
+				$return[ $id ] = false;
+				$return_cache[ $key ] = [
+					'value' => false,
+					'found' => false,
+				];
+
+				continue;
+			} else {
+				$uncached_keys[ $id ] = $key;
 			}
+		}
+
+		if ( $uncached_keys ) {
+			$this->timer_start();
+
+			$uncached_keys_list = array_values( $uncached_keys );
+			$values = $mc->get( $uncached_keys_list );
 
 			$elapsed = $this->timer_stop();
-			$this->group_ops_stats( 'get_multi', $keys, $group, null, $elapsed );
+			$this->group_ops_stats( 'get_multiple', $uncached_keys_list, $group, null, $elapsed );
+
+			foreach ( $uncached_keys as $id => $key ) {
+				$value = isset( $values[$key] ) ? $values[$key] : false;
+
+				$return[ $id ] = $value;
+				$return_cache[ $key ] = [
+					'value' => $value,
+					'found' => false !== $value,
+				];
+			}
 		}
 
 		$this->cache = array_merge( $this->cache, $return_cache );
 
 		return $return;
 	}
 
+	/**
+	 * Legacy implementation for get_multiple
+	 *
+	 * Kept here for backwards compatibility
+	 */
+	function get_multi( $groups ) {
+		/*
+		format: $get['group-name'] = array( 'key1', 'key2' );
+		*/
+		$return = array();
+
+		foreach ( $groups as $group => $ids ) {
+			$this->timer_start();
+			$result = $this->get_multiple( $ids, $group );
```

Comment:
We have some cache-key parsing in `group_ops_stats`. Reusing `get_multiple()` here would do this:
```
  1 get_multiple ["wp_:posts:3061","wp_:posts:3060"] (0.1 ms)
  2 get_multi ["061","060"] (0.1 ms)
  3 get_multi ["061","060"] (0.0 ms)
```